$NetBSD: patch-ag,v 1.17 2010/09/21 03:05:27 taca Exp $

* Add "--with-nbsdops" option, NetBSD standard options.
* Link with util(3) in the case of DragonFly, too.
* When specified "--with-kerb5" option, test existence of several functions
  even if there is krb5-config.  krb5-config dosen't give all definitions for
  functions (HAVE_KRB5_*).
* Remove setting sysconfdir to "/etc".

--- configure.orig	Mon May 16 17:38:16 2011
+++ configure
@@ -1596,7 +1596,7 @@ Fine tuning of the installation directories:
   --bindir=DIR            user executables [EPREFIX/bin]
   --sbindir=DIR           system admin executables [EPREFIX/sbin]
   --libexecdir=DIR        program executables [EPREFIX/libexec]
-  --sysconfdir=DIR        read-only single-machine data [/etc]
+  --sysconfdir=DIR        read-only single-machine data [PREFIX/etc]
   --sharedstatedir=DIR    modifiable architecture-independent data [PREFIX/com]
   --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
   --libdir=DIR            object code libraries [EPREFIX/lib]
@@ -1676,6 +1676,7 @@ Optional Packages:
   --with-libraries        additional libraries to link with
   --with-efence           link with -lefence for malloc() debugging
   --with-csops            add CSOps standard options
+  --with-nbsdops          add NetBSD standard options
   --without-passwd        don't use passwd/shadow file for authentication
   --with-skey=DIR         enable S/Key support
   --with-opie=DIR         enable OPIE support
@@ -4052,8 +4053,24 @@ $as_echo "$as_me: WARNING: Ignoring unknown argument t
 esac
 fi
 
+# Check whether --with-nbsdops or --without-nbsdops was given.
+if test "${with_nbsdops+set}" = set; then
+  withval="$with_nbsdops"
+  case $with_nbsdops in
+    yes)	echo 'Adding NetBSD standard options'
+		CHECKSIA=false
+		with_ignore_dot=yes
+		with_env_editor=yes
+		with_tty_tickets=yes
+		;;
+    no)		;;
+    *)		echo "Ignoring unknown argument to --with-nbsdops: $with_nbsdops"
+		;;
+esac
+fi;
 
 
+
 # Check whether --with-passwd was given.
 if test "${with_passwd+set}" = set; then :
   withval=$with_passwd; case $with_passwd in
@@ -14469,7 +14486,7 @@ if test "x$ac_cv_header_login_cap_h" = x""yes; then :
 _ACEOF
  LOGINCAP_USAGE='[-c class|-] '; LCMAN=1
 	case "$OS" in
-	    freebsd|netbsd)	SUDO_LIBS="${SUDO_LIBS} -lutil"
+	    dragonfly*|freebsd*|netbsd*)	SUDO_LIBS="${SUDO_LIBS} -lutil"
 	    ;;
 	esac
 
@@ -17439,6 +17456,8 @@ fi
 rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
 	AUTH_OBJS="$AUTH_OBJS kerb5.o"
     fi
+fi
+if test ${with_kerb5-'no'} != "no"; then
     _LIBS="$LIBS"
     LIBS="${LIBS} ${SUDO_LIBS}"
     for ac_func in krb5_verify_user krb5_init_secure_context
@@ -18811,7 +18830,6 @@ test "$libexecdir" = '${exec_prefix}/libexec' && libex
 test "$includedir" = '${prefix}/include' && includedir='$(prefix)/include'
 test "$datarootdir" = '${prefix}/share' && datarootdir='$(prefix)/share'
 test "$docdir" = '${datarootdir}/doc/${PACKAGE_TARNAME}' && docdir='$(datarootdir)/doc/$(PACKAGE_TARNAME)'
-test "$sysconfdir" = '${prefix}/etc' -a X"$with_stow" != X"yes" && sysconfdir='/etc'
 
 ac_config_files="$ac_config_files Makefile sudo.man visudo.man sudoers.man sudoers.ldap.man sudoreplay.man sudo_usage.h sudoers"
 
@@ -19420,6 +19438,8 @@ do
   case $ac_option in
   # Handling of the options.
   -recheck | --recheck | --rechec | --reche | --rech | --rec | --re | --r)
+	: Avoid regenerating within pkgsrc
+	exit 0
     ac_cs_recheck=: ;;
   --version | --versio | --versi | --vers | --ver | --ve | --v | -V )
     $as_echo "$ac_cs_version"; exit ;;
